subworkflows/local/quant.nf

```diff
@@ -24,60 +25,70 @@ workflow QUANT {
 
     main:
         ch_versions = Channel.empty()
-        // Rip post-percolator idXML files and manipulate such that we end up with [meta_run1, idxml_run1, pout_filtered] [meta_run2, idxml_run2, pout_filtered] ...
-        OPENMS_IDRIPPER( merged_pout ).ripped
-                .join( merge_meta_map )
-                .join( filter_q_value )
-                // TODO: fdrfiltered is not needed for idscore switching, but for idfilter. This will be adressed in the next refacoring of the workflow
-                .map { group_meta, ripped, meta, fdrfiltered -> [meta, ripped, fdrfiltered] }
-                .transpose()
-                .set { ch_ripped_pout }
-        ch_versions = ch_versions.mix(OPENMS_IDRIPPER.out.versions.ifEmpty(null))
+        // Split post-percolator idXML files and manipulate such that we end up with [meta_run1, idxml_run1] [meta_run2, idxml_run2] ...
+        // We need to make sure that the order of the runs is the same as in the mzml files since IDRipper always sorts the runs
+        // (and nextflow does not guarantee the order of the maps in merged_meta_map)
+        OPENMS_IDRIPPER( merged_pout ).idxmls
+                .flatMap { group_meta, idxmls -> idxmls.collect { idxml -> [[spectra: idxml.baseName], idxml] } }
+                // join on file basename to make sure that the order of the runs is the same as in the mzml files
+                // Is there a smoother way to do this?
+                .join( merge_meta_map
+                        .flatMap { group_meta, metas -> metas }
+                        .map { meta -> [[spectra:meta.spectra], meta]} )
+                .map { spectra, idxmls, meta -> [meta, idxmls] }
+                .set { ch_ripped_idxml }
+        ch_versions = ch_versions.mix(OPENMS_IDRIPPER.out.versions)
 
         // Switch to xcorr for filtering since q-values are set to 1 with peptide-level-fdr
         if (params.fdr_level == 'peptide_level_fdrs'){
-            ch_runs_to_be_filtered = OPENMS_IDSCORESWITCHER( ch_ripped_pout ).switched_idxml
-            ch_versions = ch_versions.mix(OPENMS_IDSCORESWITCHER.out.versions.ifEmpty(null))
+            ch_runs_score_switched = OPENMS_IDSCORESWITCHER( ch_ripped_idxml ).idxml
+            ch_versions = ch_versions.mix(OPENMS_IDSCORESWITCHER.out.versions)
         } else {
-            ch_runs_to_be_filtered = ch_ripped_pout
+            ch_runs_score_switched = ch_ripped_idxml
         }
 
+        // Manipulate such that [meta_run1, idxml_run1, pout_group1], [meta_run2, idxml_run2, pout_group1] ...
+        ch_runs_score_switched
+            // Nextflow can only combine/join on the exact groupKey object, merge_id is not sufficient
+            .map { meta, idxml -> [groupKey([id: meta.sample + '_' + meta.condition], meta.group_count) , meta, idxml] }
+            .combine(filter_q_value, by:0)
+            .map { group_meta, meta, idxml, q_value -> [meta, idxml, q_value] }
+            .set { ch_runs_to_filter}
```

Comment:
groupTuple missing?

Reply:
No, here each splitted `idxml` file needs to be filtered by the `q_value`